src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -50,21 +61,34 @@ def address_for_module(self, module: str) -> Optional[Address]:
 
 @rule
 async def map_first_party_modules_to_addresses() -> FirstPartyModuleToAddressMapping:
```

Comment:
This rule can likely be optimized for performance. For now, I wanted to get it working to give us a baseline.